src/model/src/ModelTransformers.cpp

```diff
@@ -662,10 +664,39 @@ bool createReducedModelAndChangeLinkFrames(const Model& fullModel,
 
             newJoint = (IJointPtr)newJointRevoluteSO2;
         }
+        else if (dynamic_cast<const SphericalJoint*>(oldJoint))
+        {
+            const SphericalJoint* oldJointSpherical = dynamic_cast<const SphericalJoint*>(oldJoint);
+
+            Transform oldLink1_X_oldLink2 = oldJointSpherical->getRestTransform(oldLink1, oldLink2);
+            Transform newLink1_X_newLink2 = newLink1_X_oldLink1 * oldLink1_X_oldLink2 * newLink2_X_oldLink2.inverse();
+
+            SphericalJoint* newJointSpherical = new SphericalJoint(*oldJointSpherical);
+
+            newJointSpherical->setAttachedLinks(newLink1, newLink2);
+            newJointSpherical->setRestTransform(newLink1_X_newLink2);
+
+            // Update joint centers to account for coordinate frame changes
+            // Transform center in first link (newLink1) coordinate system
+            Position oldCenterInFirstLink = oldJointSpherical->getJointCenter(oldLink1);
+            Position newCenterInFirstLink;
+            newCenterInFirstLink = newLink1_X_oldLink1.getRotation() * oldCenterInFirstLink
+                                          + newLink1_X_oldLink1.getPosition();
```

Comment:
This can be written as
```suggestion
            newCenterInFirstLink = newLink1_X_oldLink1 * oldCenterInFirstLink;
```

for compactness and make it clear what it is happening?